framework/src/service/ServiceRegistrationBase.cpp

```diff
@@ -353,12 +353,9 @@ namespace cppmicroservices
             return true;
         }
 
-        ServiceReferenceBase sr1;
-        ServiceReferenceBase sr2;
-        {
-            LockServiceRegistration(), sr1 = d->reference;
-            o.LockServiceRegistration(), sr2 = o.d->reference;
-        }
+        ServiceReferenceBase sr1 = (LockServiceRegistration(), d->reference);
+        ServiceReferenceBase sr2 = (o.LockServiceRegistration(), o.d->reference);
+
         return sr1 < sr2;
```

Comment:
the locks from `LockServiceRegistration` and `o.LockServiceRegistration` are now held while executing `operator<`. will this cause a deadlock?

Reply:
yes, you're right. I missed that.